source/views/Extension/Views/ImportWallet/components/DragDropBox.jsx

```diff
@@ -0,0 +1,141 @@
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next';
+import useStyles from '../styles';
+import { CloudUpload } from '@material-ui/icons';
+import CloseIcon from '@material-ui/icons/Close';
+import { useEffect } from 'react';
+import GradientFile from '@assets/icons/gradient-file.svg';
+import { Typography } from '@material-ui/core';
+import RedWarningIcon from '@assets/icons/red-warning-icon.svg';
+import { Plug } from "@components";
+
+const DragDropBox = ({ acceptedExtension, setLoading, setDisabled }) => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+
+  const [dragActive, setDragActive] = useState(false);
+  const [fileNotSupported, setFileNotSupported] = useState(false);
+  const [file, setFile] = useState(null);
+
+  const handleRemoveFile = () => {
+    setFile(null);
+    setDisabled(false);
+  }
+
+  useEffect(() => {
+    if (fileNotSupported) {
+      setTimeout(() => {
+        console.log('arranca setTimeOut');
+        setFileNotSupported(false);
+      }, 2000)
+    }
+  }, [fileNotSupported]);
+
+  // could be a good feature for the future accept more than one extension
+  const checkDesiredFile = (fileObject) => {
+    setLoading(true);
+    setLoading(false);
+    const fileExt = fileObject[0]?.name.split('.').pop();
+    if (fileExt === acceptedExtension) {
+      console.log('extension correcta', fileExt);
+      console.log('file extension', fileObject[0]);
+      setFile(fileObject[0]);
+      setLoading(false);
+      setDisabled(true);
+    } else {
+      setFileNotSupported(true);
+      console.log('extension incorrecta', fileExt);
+    }
+  }
+
+  const handleDrag = (e) => {
+    console.log('entra en el handle Drag');
+    e.preventDefault();
+    e.stopPropagation();
+    if (e.type === "dragenter" || e.type === "dragover") {
+      setDragActive(true);
+    } else if (e.type === "dragleave") {
+      setDragActive(false);
```

Comment:
is there any other type? If these 3 are the only ones you can avoid the second `if`